### PR TITLE
graylog-plugin-enterprise/issues/3419 backport to 4.2

### PR DESCRIPTION
**Pull request by patrickmann**
See https://github.com/Graylog2/graylog-plugin-enterprise/issues/3419 for details

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java [hunk 1]
```diff
@@ -295,6 +296,9 @@ public Response create(@ApiParam(name = "JSON body", value = "Must contain usern
             LOG.error(msg);
             throw new BadRequestException(msg);
         }
+        if (cr.roles() != null && cr.roles().contains(RoleServiceImpl.ADMIN_ROLENAME) && cr.isServiceAccount()) {
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java:299]: I think this check can be circumvented by using a rolename with different capitalization. e.g. `aDMin`

Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java:299]: the `RoleServiceImpl` loads the role with a `toLowerCase()`

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java [hunk 2]
```diff
@@ -88,6 +88,8 @@
     private final SessionResponseFactory sessionResponseFactory;
     private final ClusterConfigService clusterConfigService;
 
+    private static final String USERNAME = "username";
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java:91]: we should be using that constant also when reading the username.
e.g:
https://github.com/Graylog2/graylog2-server/blob/4.2/graylog2-server/src/main/java/org/graylog2/shared/security/ActorAwareUsernamePasswordTokenFactory.java#L29

File: graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java [hunk 3]
```diff
@@ -417,9 +423,25 @@ public void changeUser(@ApiParam(name = "userId", value = "The ID of the user to
...
+
+        if (cr.isServiceAccount() != null) {
+            user.setServiceAccount(cr.isServiceAccount());
+        }
+
         userManagementService.update(user);
     }
 
+    private void checkAdminRoleForServiceAccount(ChangeUserRequest cr, User user) {
+        if (user.isServiceAccount() && cr.roles() != null && cr.roles().contains(RoleServiceImpl.ADMIN_ROLENAME)) {
```
[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java]: fixed

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java (hunk 2 above)
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java:91]: This touches code not related to the security fix. Let's postpone that.

**Review by mpfz0r — APPROVED** [review]

**Pull request merged**